debezium-core/src/main/java/io/debezium/relational/RelationalOffsetContext.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.relational;
+
+import io.debezium.connector.SnapshotRecord;
+import io.debezium.connector.common.BaseSourceInfo;
+import io.debezium.pipeline.spi.OffsetContext;
+
+public abstract class RelationalOffsetContext implements OffsetContext {
+
+    public abstract BaseSourceInfo getSourceInfoObject();
```

Comment:
`SourceInfo` was taken and was specific to each connector so I had to create a common interface to extract it from their respective implementation classes.